internal/services/appservice/source_control_resource.go

```diff
@@ -0,0 +1,286 @@
+	RepoURL                   string                      `tfschema:"repo_url"`
+	Branch                    string                      `tfschema:"branch"`
+	LocalGitSCM               bool                        `tfschema:"use_local_git"`
+	ManualIntegration         bool                        `tfschema:"use_manual_integration"`
+	UseMercurial              bool                        `tfschema:"use_mercurial"`
+	RollbackEnabled           bool                        `tfschema:"rollback_enabled"`
+	UsesGithubAction          bool                        `tfschema:"uses_github_action"`
+	GithubActionConfiguration []GithubActionConfiguration `tfschema:"github_action_configuration"`
+}
+
+var _ sdk.Resource = AppServiceSourceControlResource{}
+
+func (r AppServiceSourceControlResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"app_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.WebAppID,
+		},
+
+		"repo_url": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			Computed:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+			RequiredWith: []string{
+				"branch",
+			},
+		},
+
+		"branch": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			Computed:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+			RequiredWith: []string{
+				"repo_url",
+			},
+		},
+
+		"use_local_git": {
+			Type:     pluginsdk.TypeBool,
+			Optional: true,
+			Default:  false,
+			ForceNew: true,
+			ConflictsWith: []string{
+				"repo_url",
+				"branch",
+				"use_manual_integration",
+				"uses_github_action",
+				"github_action_configuration",
+				"use_mercurial",
+				"rollback_enabled",
+			},
+		},
+
+		"use_manual_integration": {
+			Type:     pluginsdk.TypeBool,
+			Optional: true,
+			ForceNew: true,
+			Default:  false,
+		},
+
+		"github_action_configuration": githubActionConfigSchema(),
+
+		"use_mercurial": {
+			Type:     pluginsdk.TypeBool,
+			Optional: true,
+			ForceNew: true,
+			Default:  false,
+		},
+
+		"rollback_enabled": {
+			Type:     pluginsdk.TypeBool,
+			Optional: true,
+			ForceNew: true,
+			Default:  false,
+		},
+	}
+}
+
+func (r AppServiceSourceControlResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"scm_type": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+
+		"uses_github_action": {
+			Type:     pluginsdk.TypeBool,
+			Computed: true,
+		},
+	}
+}
+
+func (r AppServiceSourceControlResource) ModelObject() interface{} {
+	return AppServiceSourceControlModel{}
+}
+
+func (r AppServiceSourceControlResource) ResourceType() string {
+	return "azurerm_app_service_source_control" // TODO - Does this name fit the new convention?
+}
+
+func (r AppServiceSourceControlResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
```

Comment:
should we be passing this through from somewhere?

Reply:
Not to the best of my knowledge, this is the Typed SDK equivalent of `Create: pluginsdk.DefaultTimeout(30 * time.Minute)`, so should be defined here with the function rather than collected from elsewhere? Happy to discuss alternatives.